# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerTypes.ipdlh
# Commit: 77eda17d0795
# Full Hash: 77eda17d0795e32bd6bf224832b922d8e57aedca
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-04-15 21:21:06
# Regressor Bug: 1899503
# File Overlap Count: 2
# Description:
#   Bug 1899503 - P1 New top-level IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger to support nsIWorkerDebugger interface for Workers in the parent process. r=asuth
#   
#   New IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger are introduced to support remote debugger of Workers on the parent thread.
#   
#   PRemoteWorkerDebuggerManager is IPC between the parent process main thread and the content process WorkerLauncher thread.
# ==============================================================================

diff -r 358f0cf9a8d8 -r 77eda17d0795 dom/workers/remoteworkers/RemoteWorkerTypes.ipdlh
--- a/dom/workers/remoteworkers/RemoteWorkerTypes.ipdlh	Tue Apr 15 09:45:02 2025 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerTypes.ipdlh	Tue Apr 15 09:47:38 2025 +0000
@@ -21,6 +21,8 @@
 using mozilla::OriginTrials from "mozilla/OriginTrialsIPCUtils.h";
 using mozilla::RFPTargetSet from "nsRFPService.h";
 
+[RefCounted] using class nsIPrincipal from "nsIPrincipal.h";
+
 namespace mozilla {
 namespace dom {
 
@@ -136,5 +138,26 @@
   void_t;
 };
 
+
+// Followings are data structures used to for RemoteWorkerDebugger
+
+struct RemoteWorkerDebuggerInfo {
+  bool isChrome;
+  uint32_t type;
+  nsString url;
+  uint64_t windowID;
+  nsIPrincipal principal;
+  uint64_t serviceWorkerID;
+  nsString Id;
+  nsString name;
+  nsString parentId;
+};
+
+struct RemoteWorkerDebuggerErrorInfo {
+  nsCString fileName;
+  uint32_t lineNo;
+  nsString message;
+};
+
 } // namespace dom
 } // namespace mozilla